impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/changeBeanQualifier/ChangeBeanQualifierTest.java

```diff
@@ -0,0 +1,30 @@
+package org.jboss.cdi.tck.tests.build.compatible.extensions.changeBeanQualifier;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.cdi.tck.AbstractTest;
+import org.jboss.cdi.tck.shrinkwrap.WebArchiveBuilder;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.test.audit.annotations.SpecVersion;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertTrue;
+
+@SpecVersion(spec = "cdi", version = "4.0")
+public class ChangeBeanQualifierTest extends AbstractTest {
+    @Deployment
+    public static WebArchive createTestArchive() {
+        // no beans.xml + an extension = not a bean archive, bean classes are added through the extension
```

Comment:
This is actually wrong - the archive will have `beans.xml` automatically included (`WebArchiveBuilder` does that).
Instead, we could make use of the `annotated` discovery mode (just the empty beans.xml file) and simply not give those beans any BDA.